File: lib/java/src/protocol/TType.java
package com.facebook.thrift.protocol;

/**
 * Type constants in the Thrift protocol.
 *
 * @author Mark Slee <mcslee@facebook.com>
 */
public final class TType {
  public static final byte STOP   = 1;
  public static final byte BYTE   = 2;
  public static final byte U16    = 3;
  public static final byte I16    = 4;
  public static final byte U32    = 5;
  public static final byte I32    = 6;
  public static final byte U64    = 7;
  public static final byte I64    = 8;
  public static final byte STRING = 9;
  public static final byte STRUCT = 10;
  public static final byte MAP    = 11;
  public static final byte SET    = 12;
  public static final byte LIST   = 13;
}

File: lib/php/src/protocol/TType.php
<?php

/**
 * Constants for Thrift data types.
 *
 * @package thrift.protocol
 * @author Mark Slee <mcslee@facebook.com>
 */

/**
 * Data types that can be sent via Thrift
 */
class TType {
  const STOP   = 0;
  const VOID   = 1;
  const BOOL   = 2;
  const BYTE   = 3;
  const I08    = 3;
  const DOUBLE = 4;
  const I16    = 6;
  const I32    = 8;
  const I64    = 10;
  const STRING = 11;
  const UTF7   = 11;
  const STRUCT = 12;
  const MAP    = 13;
  const SET    = 14;
  const LST    = 15;    // N.B. cannot use LIST keyword in PHP!
  const UTF8   = 16;
  const UTF16  = 17;
}

/**
 * Message types for RPC
 */
class TMessageType {
  const CALL  = 1;
  const REPLY = 2;
}

?>
